Editor/SnapZone/SnapZoneSettings.cs

```diff
@@ -0,0 +1,158 @@
+using System.IO;
+using UnityEditor;
+using UnityEngine;
+using Innoactive.Creator.XRInteraction;
+
+namespace Innoactive.CreatorEditor.XRInteraction
+{
+    /// <summary>
+    /// Settings for <see cref="SnapZone"/>s for e.g. automatic creation of such snap zones.
+    /// </summary>
+    [CreateAssetMenu(fileName = "SnapZoneSettings", menuName = "Innoactive/SnapZoneSettings", order = 1)]
+    public class SnapZoneSettings : ScriptableObject
+    {
+        /// <summary>
+        /// Only Interactables with this LayerMask will interact with this <see cref="Innoactive.Creator.XRInteraction.SnapZone"/>.
+        /// </summary>
+        [Tooltip("Only Interactables with this LayerMask will interact with this SnapZone.")]
+        public LayerMask InteractionLayerMask = 1;
+
+        /// <summary>
+        /// This color is used as the snap zone highlight color when no object is hovering a <see cref="SnapZone"/>.
+        /// </summary>
+        [Tooltip("This color is used as the snap zone highlight color when no object is hovering but `Snap Zone Active` is true.")]
+        public Color HighlightColor = new Color32(64, 200, 255, 50);
+        
+        [SerializeField]
+        [Tooltip("The material used for the highlight object. Should be transparent.\n\n[This field overrides 'HighlightColor']")]
+        private Material highlightMaterial;
+
+        /// <summary>
+        /// This color is used when an <see cref="InteractableObject"/> is hovering a <see cref="SnapZone"/>.
+        /// </summary>
+        [Tooltip("This color is used when a valid object is hovering the snap zone.")]
+        public Color ValidationColor = new Color32(0, 255, 0, 50);
+
+        [SerializeField]
+        [Tooltip("The material shown when a valid object is hovering the snap zone. Should be transparent.\n\n[This field overrides 'ValidHighlightColor']")]
+        private Material validationMaterial;
+
+        /// <summary>
+        /// The material used for drawing when an <see cref="InteractableObject"/> is hovering a <see cref="SnapZone"/>. Should be transparent.
+        /// </summary>
+        public Material HighlightMaterial
+        {
+            get
+            {
+                if (highlightMaterial == null)
```

Comment:
You can move the body to SetupHighlightMaterial() method to make it more readable.